zebra-chain/src/primitives/zcash_history.rs

```diff
@@ -0,0 +1,329 @@
+//! Contains code that interfaces with the zcash_history crate from
+//! librustzcash.
+
+// TODO: remove after this module gets to be used
+#![allow(dead_code)]
+
+use std::{collections::HashMap, convert::TryInto, io, sync::Arc};
+
+use crate::{
+    block::{Block, ChainHistoryMmrRootHash},
+    parameters::{ConsensusBranchId, Network, NetworkUpgrade},
+    sapling,
+};
+
+/// A MMR Tree using zcash_history::Tree.
+///
+/// Currently it should not be used as a long-term data structure because it
+/// may grow undefinitely.
+pub struct Tree {
+    network: Network,
+    tree: zcash_history::Tree,
```

Comment:
Can we use a similar naming style to the Rust standard library?

`tree` becomes `inner_tree`
`low_node` becomes `inner_node`
`low_entry` becomes `inner_entry`
`encoded` becomes `inner` (it is clear from the type that it is encoded)